Assets/MixedRealityToolkit/UX/Scripts/Tooltip/ToolTipBackgroundBlob.cs

```diff
@@ -0,0 +1,200 @@
+            set {
+                blobRotation = Mathf.Clamp(value, 0, MaxRotation);
+            }
+        }
+
+        public float PositionCorrectionStrength {
+            get {
+                return positionCorrectionStrength;
+            }
+            set {
+                positionCorrectionStrength = Mathf.Clamp(value, MinPositionCorrection, MaxPositionCorrection);
+            }
+        }
+
+        public float DistortionCorrectionStrength {
+            get {
+                return distortionCorrectionStrength;
+            }
+            set {
+                distortionCorrectionStrength = Mathf.Clamp(value, MinDistortionCorrection, MaxDistortionCorrection);
+            }
+        }
+
+        public float RotationCorrectionStrength {
+            get {
+                return rotationCorrectionStrength;
+            }
+            set {
+                rotationCorrectionStrength = Mathf.Clamp (value, MinRotationCorrection, MaxRotationCorrection);
+            }
+        }
+
+        public Vector3 BlobOffset {
+            get {
+                return blobOffset;
+            } set {
+                blobOffset = value;
+            }
+        }
+
+        /// <summary>
+        /// Mesh renderer for mesh background.
+        /// </summary>
+        public MeshRenderer BackgroundRenderer;
+
+        [Header("Blob settings")]
+        [SerializeField]
+        [Range(0f,5f)]
+        private float blobInertia = 0.5f;
+
+        [SerializeField]
+        [Range(0f, 1f)]
+        private float blobDistortion = 0.1f;
+
+        [SerializeField]
+        [Range(0f, 1f)]
+        private float blobRotation = 0.1f;
+
+        [SerializeField]
+        [Range(0.1f, 5f)]
+        private float positionCorrectionStrength = 1f;
+
+        [SerializeField]
+        [Range(0.1f, 5f)]
+        private float distortionCorrectionStrength = 1f;
+
+        [SerializeField]
+        [Range(0.1f, 5f)]
+        private float rotationCorrectionStrength = 1f;
+
+        [SerializeField]
+        private Vector3 blobOffset;
+
+        protected override void OnEnable()
+        {
+            base.OnEnable();
+            lastPosition = PositionTarget.position;
+            inertialContentBounds = new Bounds(Vector3.zero, Vector3.one);
+            localContentBounds = new Bounds(Vector3.zero, Vector3.one);
+            velocity = Vector3.zero;
+        }
+
+        protected override void ScaleToFitContent()
+        {
+            // Get the local size of the content - this is the scale of the text under the content parent
+            Vector3 localContentSize = toolTip.LocalContentSize;
+            Vector3 localContentOffset = toolTip.LocalContentOffset;
+
+            // Get the size of the mesh and use this to adjust the local content size on the x / y axis
+            // This will accomodate meshes that aren't built to 1,1 scale
+            Bounds meshBounds = BackgroundRenderer.GetComponent<MeshFilter>().sharedMesh.bounds;
```

Comment:
If the background renderer doesn't change often we can probably cache the `MeshFilter` so we don't have to get the component each time we scale this.